src/cider/nrepl/middleware/content_type.clj

```diff
@@ -74,43 +75,44 @@
   (as-url [^URL u]
     u))
 
-(defn response+content-type
-  "Consumes an nREPL response, having a `:value`. If the `:value` is
-  recognized as an AWT Image, a File, or a File URI, rewrite the
-  response to have a `:content-type` being a MIME type of the content,
-  and a `:body` to re-use the RFC term for the message payload."
-  [{:keys [value] :as response}]
-  (cond
-    ;; FIXME (arrdem 2018-04-03):
-    ;;
-    ;;   This could be more generic in terms of tolerating more
-    ;;   protocols / schemes
-
-    ;; RFC-2017 external-body responses for UR[IL]s and things which are just wrappers thereof
-    (or (and (instance? File value)
-             (.exists ^File value))
-        (instance? URI value)
-        (instance? URL value))
-    (assoc response
-           :content-type ["message/external-body"
-                          {"access-type" "URL"
-                           "URL" (.toString ^URL (as-url value))}]
-           :body "")
-
-    ;; FIXME (arrdem 2018-04-03):
-    ;;
-    ;;   This is super snowflakey in terms of only supporting base64
-    ;;   coding this one kind of object.  This could definitely be
-    ;;   more generic / open to extension but hey at least it's
-    ;;   re-using machinery.
-
-    (instance? java.awt.Image value)
-    (with-open [bos (ByteArrayOutputStream.)]
-      (merge response
-             (when (ImageIO/write ^RenderedImage value "png" ^OutputStream bos)
-               (slurp-reply "" ["image/png" {}] (.toByteArray bos)))))
-
-    :else response))
+(defn external-body-response
+  "Partial response map having an external-body content-type referring to the given URL.
+
+  See RFC-2017: Definition of the URL MIME External-Body Access-Type."
+  [value]
+  {:content-type ["message/external-body"
+                  {"access-type" "URL"
+                   "URL" (.toString ^URL (as-url value))}]
```

Comment:
It appears `(as-url)` is only ever invoked once, and immediately followed by a .toString?

If that's the case, could the URLCoercable protocol be string-based instead?

URLs are a legacy class, right? Especially in face of its infamous side-effecting `.equals` method.

Reply:
Thanks for the suggestion, I've changed `as-url` to `as-url-string`.